lib/proxy_fetcher/utils/http_client.rb

```diff
@@ -71,17 +71,10 @@ def fetch
       response = process_http_request
       response.body.to_s
     rescue StandardError => e
-      ProxyFetcher.logger.warn("Failed to process request to #{url} (#{e.message})")
+      ProxyFetcher.config.logger.warn("Failed to process request to #{url} (#{e.message})")
```

Comment:
Difference between `ProxyFetcher.logger` and `ProxyFetcher.config.logger`.

# spec/proxy_fetcher/providers/base_spec.rb:31

```
From: /home/ydakuka/Work/proxy_fetcher/lib/proxy_fetcher/utils/http_client.rb:74 ProxyFetcher::HTTPClient#fetch:

    70: def fetch
    71:   response = process_http_request
    72:   response.body.to_s
    73: rescue StandardError => e
 => 74:   binding.pry
    75:   ProxyFetcher.logger.warn("Failed to process request to #{url} (#{e.message})")
    76:   ""
    77: end

[1] pry(#<ProxyFetcher::HTTPClient>)> ProxyFetcher.logger
=> #<ProxyFetcher::NullLogger:0x000055f8beba6b68>
[2] pry(#<ProxyFetcher::HTTPClient>)> ProxyFetcher.config.logger
=> #<Logger:0x000055f8be086be8
 @default_formatter=#<Logger::Formatter:0x000055f8be086bc0 @datetime_format=nil>,
 @formatter=nil,
 @level=0,
 @logdev=
  #<Logger::LogDevice:0x000055f8be086990
   @binmode=false,
   @dev=#<StringIO:0x000055f8be086dc8>,
   @filename=nil,
   @mon_data=#<Monitor:0x000055f8be086710>,
   @mon_data_owner_object_id=7160,
   @shift_age=nil,
   @shift_period_suffix=nil,
   @shift_size=nil>,
 @progname=nil>
[3] pry(#<ProxyFetcher::HTTPClient>)> 
```